client/packages/programs/src/JsonForms/components/Search/SearchWithUserSource.tsx

```diff
@@ -80,16 +84,19 @@ export const SearchWithUserSource = (props: ControlProps) => {
   const handleDataUpdate = (selectedResult: Record<string, any> | null) => {
     if (selectedResult === null) return;
     if (!saveFields) handleChange(path, selectedResult);
-
-    const newObj: Record<string, any> = {};
-    saveFields?.forEach(
-      field => (newObj[field] = selectedResult[field] ?? null)
-    );
-    handleChange(path, newObj);
+    else {
+      const newObj: Record<string, any> = {};
+      saveFields?.forEach(
+        field => (newObj[field] = selectedResult[field] ?? null)
+      );
+      handleChange(path, newObj);
+    }
```

Comment:
Forgot to return early from the "no saveFields condition", which caused problems, so changed to an if/else.